sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -169,8 +169,13 @@ object NestedColumnAliasing {
       projectList: Seq[NamedExpression],
       nestedFieldToAlias: Map[ExtractValue, Alias]): Seq[NamedExpression] = {
```

Comment:
Maybe we shouldn't use `Map[ExtractValue, Alias]`. We should use `Map[Expression, (ExtractValue, Alias)]` where the key is canonicalized `ExtractValue`.

Reply:
> Maybe we shouldn't use `Map[ExtractValue, Alias]`. We should use `Map[Expression, (ExtractValue, Alias)]` where the key is canonicalized `ExtractValue`.

Map[Expression,  Alias] should be enough.